app/code/BVMyBlog/Blog/Controller/Adminhtml/Post/Save.php

```diff
@@ -0,0 +1,66 @@
+<?php
+declare(strict_types=1);
+namespace BVMyBlog\Blog\Controller\Adminhtml\Post;
+
+use Magento\Framework\App\Action\HttpPostActionInterface;
+
+/**
+ * Class Save
+ *
+ * Controller Save
+ */
+class Save extends \Magento\Backend\App\Action implements HttpPostActionInterface
+{
+    private $resultPageFactory;
+    private $postFactory;
+
+    /**
+     * Construct
+     *
+     * @param \Magento\Backend\App\Action\Context $context
+     * @param \Magento\Framework\View\Result\PageFactory $resultPageFactory
+     * @param \BVMyBlog\Blog\Model\PostFactory $postFactory
+     */
+    public function __construct(
+        \Magento\Backend\App\Action\Context $context,
+        \Magento\Framework\View\Result\PageFactory $resultPageFactory,
+        \BVMyBlog\Blog\Model\PostFactory $postFactory
+    ) {
+        $this->resultPageFactory = $resultPageFactory;
+        $this->postFactory = $postFactory;
+        parent::__construct($context);
+    }
+
+    /**
+     * Function execute
+     */
+    public function execute()
+    {
+        $resultRedirect = $this->resultRedirectFactory->create();
+        $data = $this->getRequest()->getPostValue();
+
+        if ($data) {
+            try {
+                $id = $data['post_id'];
+                $image_url = $data['url_key'][0]['url'];
+                $data['url_key'] = $image_url;
```

Comment:
What are you doing here? If it is possible please correct data from storefront to avoid such movements. If not please explain why